### ISSUE TITLE
Parameter expansion plan for 1.9

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#3257]
**Oscillators:** 7 more parameters (for whatever)

**Mixer:** Currently has 25 parameters. Add osc 4-6 volume (3), osc 4-6 routing (3), osc 4-6 solo (3), osc 4-6 mute (3), all channels pan (9), ring mod source selectors (4), pre/post filter mixer toggle (non-param), post filter mix gain (1), repeat the mixer sans the routing buttons and ring mod source selectors for post mixer (25 parameters). Total 51 more parameters.

**FM Routing:** Add FM matrix mode switch (non-param), add 6x6 matrix for FM depths (36 more parameters). Worse case scenario, add lower triangle matrix (21 more parameters).

**Filters:** Add 2 more parameters per filter (4 more parameters total).

**Effects:** 24 more parameters (for whatever).

**Effect slots:** Add 2 more effect slots per scene, and 2 more global effect slots.

**Comment by mkruselj** [comment]
Further thoughts:

* Oscillators will need unison stereo spread parameter
* Oscillators will need start phase parameter
* Oscillators will need Character parameter (#3338)
* Mixer routing should be continuous between F1 and F2, instead of three position integer (knob or small slider!)
* Skin engine would need to be updated so that we can define panels containing parameters, so that we could create layouts for each individual oscillator type (i.e. reorder sliders so that Skew Horizontal is right next to Skew Vertical for WT oscillator, say). This would also be used by the effects, as well

**Comment by baconpaul** [comment]
I really like this document
14 per osc and 20 per fx seem a lot but otherwise spot on. My guess is moderation may walk us back from that a bit 
I specially like using the skin engine for per oscillator type layout though too

And don’t forget that this will couple with knobs! Which is already an issue

**Comment by baconpaul** [comment]
But man with all those params you could make a pretty amazing spectral oscillator

**Comment by mkruselj** [comment]
Yeah my thinking is better expand the count so that we don't need to expand it urgently later, which would wreak havoc in automation list ordering for DAWs that don't do VST3 clumping (Reaper, Reason, Live...)

**Comment by baconpaul** [comment]
Yeah sure. Some number will feel right. The flip side is we don’t want to expose hundreds of ‘do nothing’ params to the DAW. But it’s a balance I’m sure we will get when we do it.

**Comment by mkruselj** [comment]
> 14 per osc

@baconpaul I meant 14 _more_ per osc. So 21 total for osc. [FM4 osc](https://github.com/surge-synthesizer/surge/issues/3619) will need at least 14 if not more. :)
